modules/kinesis/src/it/scala/com/snowplowanalytics/snowplow/enrich/kinesis/EnrichKinesisSpec.scala

```diff
@@ -0,0 +1,110 @@
+ */
+package com.snowplowanalytics.snowplow.enrich.kinesis
+
+import scala.concurrent.duration._
+
+import cats.effect.{IO, Resource}
+
+import retry.syntax.all._
+import retry.RetryPolicies
+
+import cats.implicits._
+
+import cats.effect.testing.specs2.CatsIO
+
+import org.testcontainers.containers.{GenericContainer => JGenericContainer}
+
+import org.specs2.mutable.Specification
+import org.specs2.specification.AfterAll
+
+class EnrichKinesisSpec extends Specification with AfterAll with CatsIO {
+
+  override protected val Timeout = 10.minutes
+
+  val retryPolicy =
+    RetryPolicies.limitRetriesByCumulativeDelay(5.minutes, RetryPolicies.capDelay[IO](2.second, RetryPolicies.fullJitter[IO](1.second)))
+
+  def afterAll: Unit = Containers.localstack.stop()
+
+  "enrich-kinesis" should {
+    "initialize with the minimal config" in {
+      Resource.make[IO, JGenericContainer[_]](
+        IO(Containers.enrich(configPath = "config/config.kinesis.minimal.hocon", loggerName = "minimal"))
+      )(
+        e => IO(e.stop())
+      ).use { e =>
+        IO(e.getLogs must contain("Running Enrich"))
+      }
+    }
+
+    "emit the correct number of enriched events and bad rows" in {
+      import count.Helpers._
+
+      val nbGood = 100l
+      val nbBad = 10l
+
+      Resource.make[IO, JGenericContainer[_]](
+        IO(
+          Containers.enrich(
+            configPath = "modules/kinesis/src/it/resources/enrich/count.hocon",
+            loggerName = "count",
+            network = Some(Containers.localstackNetwork)
+          )
+        )
+      )(
+        e => IO(e.stop())
+      ).use { _ =>
+        for {
+          aggregates <- mkResources[IO](nbGood, nbBad, localstackPort = Containers.localStackPort).use { r =>
+            r.readOutput.concurrently(r.writeInput).compile.toList
+          }
+          good = aggregates.collect { case OutputRow.Good(e) => e}
+          bad = aggregates.collect { case OutputRow.Bad(b) => b}
+          _ <- IO(println(s"Bad rows:\n${bad.map(_.compact).mkString("\n")}"))
+        } yield {
+          good.size.toLong === nbGood
+          bad.size.toLong === nbBad
+        }
+      }
+    }
+
+    "shutdown when it receives a SIGTERM" in {
+      Resource.make[IO, JGenericContainer[_]](
+        IO(
+          Containers.enrich(
+            configPath = "modules/kinesis/src/it/resources/enrich/stop.hocon",
+            loggerName = "stop",
+            network = Some(Containers.localstackNetwork)
+          )
+        )
+      )(
+        e => IO(e.stop()) // in case SIGTERM failed to stop the container
+      ).use { enrich =>
+        for {
+          _ <- IO.sleep(20.seconds) // give Enrich more time to initialize (Kinesis)
+          _ <- IO(println("stop - Sending signal"))
+          _ <- IO(enrich.getDockerClient().killContainerCmd(enrich.getContainerId()).withSignal("TERM").exec())
+          _ <- IO(enrich.isRunning()).retryingOnFailures(
```

Comment:
Let's extract this some method like `waitUntilStopped`. Retry policy could be also moved there maybe.